package.json

```diff
@@ -2,63 +2,112 @@
-    "build": "npm run build:js && npm run build:types",
-    "build:js": "ipjs build --tests --main && npm run build:copy",
-    "build:copy": "mkdir -p dist/examples/ && cp -a tsconfig.json .npmignore *.js *.ts lib test dist/ && cp examples/*.* dist/examples/ && rm -rf dist/test/fixtures/",
-    "build:types": "tsc --build && mv types dist",
-    "test:cjs": "rm -rf dist && npm run build && cp test/go.car dist/cjs/node-test/ && mocha dist/cjs/node-test/test-*.js && mocha dist/cjs/node-test/node-test-*.js && npm run test:cjs:browser",
-    "test:esm": "rm -rf dist && npm run build && cp test/go.car dist/esm/node-test/ && mocha dist/esm/node-test/test-*.js && mocha dist/esm/node-test/node-test-*.js && npm run test:esm:browser",
-    "test:node": "c8 --check-coverage --branches 100 --functions 100 --lines 100 --exclude lib/header-validator.js --exclude test/ mocha test/test-*.js test/node-test-*.js",
-    "test:cjs:browser": "polendina --page --worker --serviceworker --cleanup dist/cjs/browser-test/test-*.js",
-    "test:esm:browser": "polendina --page --worker --serviceworker --cleanup dist/esm/browser-test/test-*.js",
-    "test": "npm run lint && npm run test:node && npm run test:cjs && npm run test --prefix examples/",
-    "test:ci": "npm run lint && npm run test:node && npm run test:esm && npm run test:cjs && npm run test --prefix examples/",
-    "coverage": "c8 --reporter=html --reporter=text mocha test/test-*.js && npx st -d coverage -p 8888",
-    "docs": "jsdoc4readme --readme --description-only lib/reader*.js lib/indexed-reader.js lib/iterator.js lib/indexer.js lib/writer*.js lib/buffer-writer.js lib/decoder.js"
+  "author": "Rod <rod@vagg.org> (http://r.va.gg/)",
+  "license": "Apache-2.0 OR MIT",
+  "homepage": "https://github.com/ipld/js-car#readme",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/ipld/js-car.git"
+  },
+  "bugs": {
+    "url": "https://github.com/ipld/js-car/issues"
   },
   "keywords": [
     "car",
-    "ipld",
     "ipfs",
+    "ipld",
     "multiformats"
   ],
-  "author": "Rod <rod@vagg.org> (http://r.va.gg/)",
-  "license": "(Apache-2.0 AND MIT)",
+  "engines": {
+    "node": ">=16.0.0",
+    "npm": ">=7.0.0"
+  },
+  "type": "module",
+  "types": "./dist/src/index.d.ts",
+  "typesVersions": {
+    "*": {
+      "*": [
+        "*",
+        "dist/*",
+        "dist/src/*",
+        "dist/src/*/index"
+      ],
+      "src/*": [
+        "*",
+        "dist/*",
+        "dist/src/*",
+        "dist/src/*/index"
+      ]
+    }
+  },
+  "files": [
+    "src",
+    "dist",
+    "!dist/test",
+    "!**/*.tsbuildinfo"
+  ],
   "exports": {
     ".": {
-      "browser": "./car-browser.js",
-      "import": "./car.js"
+      "types": "./dist/src/index.d.ts",
+      "import": "./src/index.js",
+      "browser": "./src/index-browser.js"
     },
-    "./decoder": {
-      "import": "./lib/decoder.js"
+    "./buffer-writer": {
+      "types": "./src/buffer-writer.d.ts",
```

Comment:
I think these all need `./dist` in them, from this one downward; I don't believe that's taken care of by `typesVersions` mapping is it?

Reply:
Yes - have updated it.